MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SpinningSymbolLayerIconActivity.java

```diff
@@ -0,0 +1,180 @@
+
+import android.animation.ValueAnimator;
+import android.os.Bundle;
+import android.view.animation.LinearInterpolator;
+
+import com.mapbox.mapboxandroiddemo.R;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+import com.mapbox.mapboxsdk.maps.Style;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+import com.mapbox.mapboxsdk.style.sources.GeoJsonSource;
+import com.mapbox.mapboxsdk.utils.BitmapUtils;
+
+import java.io.InputStream;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import timber.log.Timber;
+
+import static com.mapbox.mapboxsdk.style.layers.Property.ICON_PITCH_ALIGNMENT_MAP;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconAllowOverlap;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconPitchAlignment;
+import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.iconRotate;
+
+/**
+ * Use an Android-system ValueAnimator and the Maps SDK's runtime styling functionality to
+ * continually updated a SymbolLayer icons' rotation. This creates a spinning effect.
+ */
+public class SpinningSymbolLayerIconActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+  private static final String SOURCE_ID = "SOURCE_ID";
+  private static final String ICON_ID = "ICON_ID";
+  private static final String LAYER_ID = "LAYER_ID";
+  private static final int DESIRED_SECONDS_PER_ONE_FULL_360_SPIN = 1;
+  private MapView mapView;
+  private MapboxMap mapboxMap;
+  private ValueAnimator iconSpinningAnimator;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_spinning_symbol_layer_icon);
+
+    // Initialize the MapView
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+
+    // Create a Style object
+    mapboxMap.setStyle(new Style.Builder().fromUri(Style.MAPBOX_STREETS)
+
+      // Add the SymbolLayer icon image to the map style
+      .withImage(ICON_ID, BitmapUtils.getBitmapFromDrawable(
+        getResources().getDrawable(R.drawable.hurricane_icon)))
+
+      // Adding a GeoJson source for the SymbolLayer icons.
+      .withSource(new GeoJsonSource(SOURCE_ID, loadGeoJsonFromAsset("spinning_icon.geojson")))
```

Comment:
As a source constructor, you can use `GeoJsonsource(String, URI)` and pass  `asset://spinning_icon.geojson` instead of loading the JSON manually.

Reply:
Fixed